actions/river/delete.php

```diff
@@ -6,14 +6,23 @@
  * @subpackage Core
  */
 
-$id = get_input('id', false);
+$id = (int)get_input('id');
 
-if ($id !== false && elgg_is_admin_logged_in()) {
-	if (elgg_delete_river(array('id' => $id))) {
-		system_message(elgg_echo('river:delete:success'));
-	} else {
-		register_error(elgg_echo('river:delete:fail'));
-	}
+$items = elgg_get_river(['id' => $id]);
+if (!$items) {
+	register_error(elgg_echo('river:delete:fail'));
+	forward(REFERER);
+}
+
+$item = $items[0];
+
+if (!$item->canDelete()) {
+	register_error(elgg_echo('river:delete:lack_permission'));
+	forward(REFERER);
+}
+
+if (elgg_delete_river(array('id' => $item->id))) {
```

Comment:
It isn't but you have to slice an array to get an item
On Sep 16, 2015 4:57 PM, "Steve Clay" <notifications@github.com> wrote:

> In actions/river/delete.php
> <https://github.com/Elgg/Elgg/pull/8942#discussion_r39640167>:
>
> > -		register_error(elgg_echo('river:delete:fail'));
> > -	}
> > +$items = elgg_get_river(['id' => $id]);
> > +if (!$items) {
> > +	register_error(elgg_echo('river:delete:fail'));
> > +	forward(REFERER);
> > +}
> > +
> > +$item = $items[0];
> > +
> > +if (!$item->canDelete()) {
> > +	register_error(elgg_echo('river:delete:lack_permission'));
> > +	forward(REFERER);
> > +}
> > +
> > +if (elgg_delete_river(array('id' => $item->id))) {
>
> elgg_get_river(['id' => $id]) isn't too bad.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8942/files#r39640167>.
>